csrc/scheduler/normalization_utils.cpp

```diff
@@ -845,8 +845,15 @@ int64_t getMaxRegOrSharedMemorySizeForPersistentBuffer(
   return available_persistent_buffer_size;
 }
 
-// Returns true if the gains of reducing buffer size is larger than the pains of
-// recalculations. We don't know the real answer until we run it.
+//  Evaluates whether to project a persistent buffer to inputs.
```

Comment:
Most useful comments do not just describe what the corresponding code does but give the important context and reasoning for what the code does. For example, here, it would be more informative to have something like:

```
Returns true if persistent buffers are projected to inputs, meaning the inputs are cached instead of the persistent buffers.

The decision of projection is primarily based on the required sizes of the two cases -- projection is done if projecting to the inputs results in a smaller size. However, we experimentally found that certain relatively expensive operations should not be projected even when that would require a larger buffer size. Specifically,

- rng: should never be projected no matter how much larger the buffer would consume
- exp in inner normalization: only allowed to get projected if the buffer is smaller than a certain size

Otherwise, as long as the projected inputs are smaller than the original persistent buffers, this function returns `true`.



```

Reply:
will follow this style